icu4j/main/tests/core/src/com/ibm/icu/dev/test/charsetdet/TestCharsetDetector.java

```diff
@@ -1249,5 +1250,55 @@ public void run() {
         }
     }
 
+    @Test
+    public void plainAsciiLeadsToAsciiBeingFirst() throws Exception {
+        // Content: "@{,xxx-yyy-!!!}"
+        // Non-English text because otherwise, it is detected as ISO-8859-1
+        byte[] input = new byte[]{0x40, 0x7b, 0x2c, 0x78, 0x78, 0x78, 0x2d, 0x79, 0x79, 0x79, 0x2d, 0x21, 0x21, 0x21, 0x7d};
+
+        CharsetDetector charsetDetector = new CharsetDetector();
+        charsetDetector.setText(input);
+        CharsetMatch[] matches = charsetDetector.detectAll();
+
+        // "just" check that ASCII is first
+        Assert.assertEquals("ASCII", matches[0].getName());
+        Assert.assertEquals(35, matches[0].getConfidence());
```

Comment:
This looks brittle. Please test greater-or-equal. See also my implementation comment about the confidence.